n2survey/lime/survey.py

```diff
@@ -1235,6 +1269,100 @@ def get_choices(self, question: str) -> dict:
 
         return choices_dict
 
+    def rate_supervision(
+        self,
+        question: str,
+        condition: str = None,
+        keep_subscores: bool = False,
+    ) -> pd.DataFrame:
+        """Calculate average direct/formal supervision rating
+
+        Args:
+            question (str): Question ID to use for calculation
+            condition (str, optional): Which kind of supervision condition to rate,
+                "formal_supervision", "direct_supervision". If not specified,
+                the condition is automatically infered. Default None.
+            keep_subscores (bool, optional): Whether to include scores from subquestions
+                in the output DataFrame, or only total score and classification.
+                Default False.
+
+        Returns:
+            pd.DataFrame: Supervision ratings and classifications
+        """
+        question_label = self.get_label(question)
+        # Infer condition type if not provided
```

Comment:
Yes, implemented.
However, using only the `question_label` for inferring the `label` will kind of make the additional input of the dict in `transform_question` obsolete.
Meaning that `transformation_questions={
        "formal_supervision": "E7a",
        "direct_supervision": "E7a"}` will both times only yield formal_supervison_scores/formal_supervision_labels (because the same question is specified).

I tried to just use one dict entry, `{'supervision', 'supervision'}` in `transform_question`, but then it is not possible to do both transformations (formal, direct) with one `read_responses` call. I could also be missing something.

Reply:
> Meaning that transformation_questions={ "formal_supervision": "E7a", "direct_supervision": "E7a"} will both times only yield formal_supervison_scores/formal_supervision_labels (because the same question is specified).

I think this is fine, because one question should only have one correct transformation. In the end, as far as I'm aware, all these transformation questions should be processed automatically, and not by user input, so it would not be necessary to account for erroneous user input. 

> I tried to just use one dict entry, {'supervision', 'supervision'} in transform_question, but then it is not possible to do both transformations (formal, direct) with one read_responses call.

I see what you mean. I would say that it is perfectly fine to leave it as is, although a little bit redundant but at least clear enough. Personally I think the optimal solution would be to modify the `read_responses` method to allow a list of questions needing the same type of transformation, e.g. `transformation_questions={"supervision": ["E7a", "E7b"]}`. But I will create a separate issue for it later.